File: Exemplos/App_eXperience_ReactNative/Elgin_AppExperience_ReactNative-font/android/app/src/main/java/com/elginm8/Printer.java
package com.elginm8;

import android.app.Activity;
import android.graphics.Bitmap;
import android.util.Base64;
import android.graphics.BitmapFactory;

import com.elgin.e1.Impressora.Termica;

import java.io.File;
import com.facebook.react.bridge.ReadableMap;
import com.facebook.systrace.SystraceMessage;

public class Printer {
    private static Activity mActivity;

    public static boolean isPrinterInternSelected;

    public Printer(Activity activity) {
        this.mActivity = activity;
        Termica.setContext(this.mActivity);
    }

    public static int printerInternalImpStart() {
        printerStop();
        int result = Termica.AbreConexaoImpressora(6, "M8", "", 0);

        if (result == 0) isPrinterInternSelected = true;

        return result;
    }

    public static int printerExternalImpIpStart(ReadableMap map) {
        printerStop();
        String model = (String) map.getString("printerModel");
        String ip = (String) map.getString("ip");
        int port = (Integer) map.getInt("port");


        try {
            int result = Termica.AbreConexaoImpressora(3, model, ip, port);
            System.out.println("result EXTERNA IP: " + result);
            if (result == 0) isPrinterInternSelected = false;

            return result;
        } catch (Exception e) {
            System.out.println("exception: " + e);
            return printerInternalImpStart();
        }

    }

    public static int printerExternalImpUsbStart(ReadableMap map) {
        printerStop();

        String model = (String) map.getString("printerModel");


        try {
            int result = Termica.AbreConexaoImpressora(1, model, "USB", 0);
            if (result == 0) isPrinterInternSelected = false;

            return result;
        } catch (Exception e) {

            return printerInternalImpStart();
        }
    }

    public static void printerStop() {
        Termica.FechaConexaoImpressora();
    }

    public static int AvancaLinhas(ReadableMap map) {
        int lines = (Integer) map.getInt("quant");

        return Termica.AvancaPapel(lines);
    }

    public static int cutPaper(ReadableMap map) {
        int lines = (Integer) map.getInt("quant");
        return Termica.Corte(lines);
    }

    public static int imprimeTexto(ReadableMap map) {
        String text = (String) map.getString("text");
        String align = (String) map.getString("align");
        String font = (String) map.getString("font");
        int fontSize = (Integer) map.getInt("fontSize");
        boolean isBold = (boolean) map.getBoolean("isBold");
        boolean isUnderline = (boolean) map.getBoolean("isUnderline");

        int result;

        int alignValue = 0;
        int styleValue = 0;

        // ALINHAMENTO VALUE
        if (align.equals("Esquerda")) {
            alignValue = 0;
        } else if (align.equals("Centralizado")) {
            alignValue = 1;
        } else {
            alignValue = 2;
        }
        //STILO VALUE
        if (font.equals("FONT B")) {
            styleValue += 1;
        }
        if ((boolean) map.getBoolean("isUnderline")) {
            styleValue += 2;
        }
        if ((boolean) map.getBoolean("isBold")) {
            styleValue += 8;
        }

        result = Termica.ImpressaoTexto(text, alignValue, styleValue, fontSize);
        return result;
    }

    private static int codeOfBarCode(String barCodeName) {
        if (barCodeName.equals("UPC-A"))
            return 0;
        else if (barCodeName.equals("UPC-E"))
            return 1;
        else if (barCodeName.equals("EAN 13") || barCodeName.equals("JAN 13"))
            return 2;

        else if (barCodeName.equals("EAN 8") || barCodeName.equals("JAN 8"))
            return 3;
        else if (barCodeName.equals("CODE 39"))
            return 4;
        else if (barCodeName.equals("ITF"))
            return 5;
        else if (barCodeName.equals("CODE BAR"))
            return 6;
        else if (barCodeName.equals("CODE 93"))
            return 7;
        else if (barCodeName.equals("CODE 128"))
            return 8;
        else return 0;
    }

    public static int imprimeBarCode(ReadableMap map) {
        int barCodeType = codeOfBarCode((String) map.getString("barCodeType"));
        String text = (String) map.getString("text");
        int height = (Integer) map.getInt("height");
        int width = (Integer) map.getInt("width");
        String align = (String) map.getString("align");

        int hri = 4; // NO PRINT
        int result;
        int alignValue;

        if (align.equals("Esquerda")) {
            alignValue = 0;
        } else if (align.equals("Centralizado")) {
            alignValue = 1;
        } else {
            alignValue = 2;
        }

        Termica.DefinePosicao(alignValue);

        result = Termica.ImpressaoCodigoBarras(barCodeType, text, height, width, hri);
        return result;
    }

    public static int imprimeQR_CODE(ReadableMap map) {
        int size = (Integer) map.getInt("qrSize");
        String text = (String) map.getString("text");
        String align = (String) map.getString("align");

        int nivelCorrecao = 2;
        int result;
        int alignValue;

        if (align.equals("Esquerda")) {
            alignValue = 0;
        } else if (align.equals("Centralizado")) {
            alignValue = 1;
        } else {
            alignValue = 2;
        }

        Termica.DefinePosicao(alignValue);

        result = Termica.ImpressaoQRCode(text, size, nivelCorrecao);
        return result;
    }

    public static int imprimeImagem(ReadableMap map) {

        int result=-1;

        System.out.println("IMPRIMINDO IMAGEM");
        String pathImage = (String) map.getString("pathImage");
        boolean isBase64 = (boolean) map.getBoolean("isBase64");



        File mSaveBit = new File(pathImage); // Your image file

        Bitmap bitmap;

        if (pathImage.equals("elgin")) {
            int id = 0;

            id = mActivity.getApplicationContext().getResources().getIdentifier(
                    pathImage,
                    "drawable",
                    mActivity.getApplicationContext().getPackageName()
            );
            System.out.println("id: " + id);

            bitmap = BitmapFactory.decodeResource(mActivity.getApplicationContext().getResources(), id);
        } else {
            if (isBase64) {
                byte[] decodedString = Base64.decode(pathImage, Base64.DEFAULT);
                bitmap = BitmapFactory.decodeByteArray(decodedString, 0, decodedString.length);

            } else {
                String filePath = mSaveBit.getPath();
                bitmap = BitmapFactory.decodeFile(filePath);
            }
        }


        if (isPrinterInternSelected) result = Termica.ImprimeBitmap(bitmap);
        else {
            result=Termica.ImprimeImagem(bitmap);
        }
        return result;

    }


    public static int imprimeXMLNFCe(ReadableMap map){
        String xmlNFCe = (String) map.getString("xmlNFCe");
        int indexcsc = (int) map.getInt("indexcsc");
        String csc = (String) map.getString("csc");
        int param = (int) map.getInt("param");
        return Termica.ImprimeXMLNFCe(xmlNFCe, indexcsc, csc, param);
    }

    public static int imprimeXMLSAT(ReadableMap map){
        String xml = (String) map.getString("xmlSAT");
        int param = (int) map.getInt("param");
        return Termica.ImprimeXMLSAT(xml, param);
    }

    public static int imprimeCupomTEF(ReadableMap map){
        String base64 = (String) map.getString("base64");

        return Termica.ImprimeCupomTEF(base64);
    }

    public static int statusGaveta(){
        return Termica.StatusImpressora(1);
    }

    public static int abrirGaveta() { return Termica.AbreGavetaElgin(); }

    public static int statusSensorPapel(){
        return Termica.StatusImpressora(3);
    }
}

File: Exemplos/App_eXperience_Kotlin/app/src/main/java/com/example/e1_elgin_kotlin/Printer.kt
package com.example.e1_elgin_kotlin

import android.app.Activity
import android.graphics.Bitmap
import com.elgin.e1.Impressora.Termica

class Printer(var mActivity: Activity) {

    fun printerInternalImpStart(): Int {
        printerStop()
        return Termica.AbreConexaoImpressora(6, "M8", "", 0)
    }

    fun printerExternalImpStart(ip: String?, port: Int): Int {
        printerStop()
        return try {
            Termica.AbreConexaoImpressora(3, "I9", ip, port)
        } catch (e: Exception) {
            printerInternalImpStart()
        }
    }

    fun printerStop() {
        Termica.FechaConexaoImpressora()
    }

    fun AvancaLinhas(map: Map<*, *>): Int {
        val lines = map["quant"] as Int
        return Termica.AvancaPapel(lines)
    }

    fun cutPaper(map: Map<*, *>): Int {
        val lines = map["quant"] as Int
        return Termica.Corte(lines)
    }

    fun imprimeTexto(map: Map<*, *>): Int {
        val text = map["text"] as String?
        val align = map["align"] as String?
        val font = map["font"] as String?
        val fontSize = map["fontSize"] as Int

        //System.out.println(text + " " + align + " " + font + " " + String.valueOf(fontSize));
        var result = 0
        var alignValue = 0
        var styleValue = 0

        // ALINHAMENTO VALUE
        alignValue = if (align == "Esquerda") {
            0
        } else if (align == "Centralizado") {
            1
        } else {
            2
        }
        //STILO VALUE
        if (font == "FONT B") {
            styleValue += 1
        }
        if (map["isUnderline"] as Boolean) {
            styleValue += 2
        }
        if (map["isBold"] as Boolean) {
            styleValue += 8
        }
        result = Termica.ImpressaoTexto(text, alignValue, styleValue, fontSize)
        return result
    }

    private fun codeOfBarCode(barCodeName: String?): Int {
        return if (barCodeName == "UPC-A") 0 else if (barCodeName == "UPC-E") 1 else if (barCodeName == "EAN 13" || barCodeName == "JAN 13") 2 else if (barCodeName == "EAN 8" || barCodeName == "JAN 8") 3 else if (barCodeName == "CODE 39") 4 else if (barCodeName == "ITF") 5 else if (barCodeName == "CODE BAR") 6 else if (barCodeName == "CODE 93") 7 else if (barCodeName == "CODE 128") 8 else 0
    }

    fun imprimeBarCode(map: Map<*, *>): Int {
        val barCodeType = codeOfBarCode(map["barCodeType"] as String?)
        val text = map["text"] as String?
        val height = map["height"] as Int
        val width = map["width"] as Int
        val align = map["align"] as String?
        val hri = 4 // NO PRINT
        val result: Int
        val alignValue: Int

        // ALINHAMENTO VALUE
        alignValue = if (align == "Esquerda") {
            0
        } else if (align == "Centralizado") {
            1
        } else {
            2
        }
        Termica.DefinePosicao(alignValue)
        result = Termica.ImpressaoCodigoBarras(barCodeType, text, height, width, hri)
        return result
    }

    fun imprimeQR_CODE(map: Map<*, *>): Int {
        val size = map["qrSize"] as Int
        val text = map["text"] as String?
        val align = map["align"] as String?
        val nivelCorrecao = 2
        val result: Int
        val alignValue: Int

        // ALINHAMENTO VALUE
        alignValue = if (align == "Esquerda") {
            0
        } else if (align == "Centralizado") {
            1
        } else {
            2
        }
        Termica.DefinePosicao(alignValue)
        result = Termica.ImpressaoQRCode(text, size, nivelCorrecao)
        return result
    }

    fun imprimeImagem(bitmap: Bitmap?): Int {


//        String pathImage = (String) map.get("pathImage");
//        boolean isBase64 = (boolean) map.get("isBase64");
//
        val result: Int
        //
//        File mSaveBit = new File(pathImage); // Your image file
//        Bitmap bitmap;
//
//        if(isBase64){
//            byte[] decodedString = Base64.decode(pathImage, Base64.DEFAULT);
//            bitmap = BitmapFactory.decodeByteArray(decodedString, 0, decodedString.length);
//
//        }else{
//            String filePath = mSaveBit.getPath();
//            bitmap = BitmapFactory.decodeFile(filePath);
//        }
        result = Termica.ImprimeBitmap(bitmap)
        return result
    }

    fun imprimeXMLNFCe(map: Map<*, *>): Int {
        val xmlNFCe = map["xmlNFCe"] as String?
        println(xmlNFCe)
        val indexcsc = map["indexcsc"] as Int
        val csc = map["csc"] as String?
        val param = map["param"] as Int
        return Termica.ImprimeXMLNFCe(xmlNFCe, indexcsc, csc, param)
    }

    fun imprimeXMLSAT(map: Map<*, *>): Int {
        val xml = map["xmlSAT"] as String?
        val param = map["param"] as Int
        return Termica.ImprimeXMLSAT(xml, param)
    }

    fun statusGaveta(): Int {
        return Termica.StatusImpressora(1)
    }

    fun abrirGaveta(): Int {
        return Termica.AbreGavetaElgin()
    }

    fun statusSensorPapel(): Int {
        return Termica.StatusImpressora(3)
    }

    fun imprimeCupomTEF(map: Map<*, *>): Int {
        val base64 = map["base64"] as String?

        return Termica.ImprimeCupomTEF(base64)
    }

    init {
        Termica.setContext(mActivity)
    }
}